lib/ABI/FunctionType.cpp

```diff
@@ -169,24 +175,31 @@ class ConversionHelper {
 
     model::RawFunctionType Result;
     Result.CustomName = Function.CustomName;
+    Result.OriginalName = Function.OriginalName;
 
+    model::StructType StackArguments;
+    uint64_t CombinedStackArgumentSize = 0;
     for (size_t ArgIndex = 0; ArgIndex < Arguments.size(); ++ArgIndex) {
       auto &ArgumentStorage = Arguments[ArgIndex];
       const auto &ArgumentType = Function.Arguments.at(ArgIndex).Type;
       if (!ArgumentStorage.Registers.empty()) {
         // Handle the registers
-        auto OriginalName = Function.Arguments.at(ArgIndex).CustomName;
+        auto ArgumentName = Function.Arguments.at(ArgIndex).name();
         for (size_t Index = 0; auto Register : ArgumentStorage.Registers) {
-          auto FinalName = OriginalName;
-          if (ArgumentStorage.Registers.size() > 1 && !FinalName.empty())
-            FinalName += "_part_" + std::to_string(++Index) + "_out_of_"
-                         + std::to_string(ArgumentStorage.Registers.size());
           model::NamedTypedRegister Argument(Register);
           Argument.Type = chooseArgumentType(ArgumentType,
                                              Register,
                                              ArgumentStorage.Registers,
                                              TheBinary);
-          Argument.CustomName = FinalName;
+
+          std::string TemporaryName = ArgumentName.str().str();
+          if (ArgumentStorage.Registers.size() > 1 && !TemporaryName.empty())
+            TemporaryName = "part_of_" + std::move(TemporaryName) + "_"
+                            + std::to_string(++Index) + "_out_of_"
+                            + std::to_string(ArgumentStorage.Registers.size());
+          if (llvm::StringRef{ TemporaryName }.take_front(8) != "unnamed_")
+            Argument.CustomName = TemporaryName;
```

Comment:
> What's going on here? 

When an argument has to be split, I prefer to "mark" it in some what to clearly see where it comes from.

> We never generate `CustomNames`.

Should I just write it into `OriginalName` then?

Reply:
No, as a policy, we never commit to the model any generated name. They are tied to a specific version of revng and if they look bad we can't change them because they're in the model.

If you want to have nice names, we need a function that computes them dynamically reasoning on the type system.
But we never commit generated names to the model.

Reply:
I'm never relying on this name being there or anything. It's there for human readers. If, say, we have a function
```cpp
void function(my_cool_struct this_name_is_important_to_me);
```
but we have no choice but to split the struct into a couple of `int`s, I'd rather it become
```cpp
void function(uint64_t part_of_this_name_is_important_to_me_1_out_of_2,
              uint64_t part_of_this_name_is_important_to_me_2_out_of_2);
```
than something nameless.

Reply:
What matters to me is that these "parts" can still be traceable - nothing more. As such "OriginalName` works.

Reply:
Done: https://github.com/revng/revng/pull/148/commits/91b7d815be9d8bbf0a526577804567ddb991e229.